sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheck.java

```diff
@@ -0,0 +1,215 @@
+ * the exception to the junit caller which will display the full exception to the user.
+ * </p>
+ * <p>
+ * An example of how to configure the check is:
+ * </p>
+ * <pre>
+ * &lt;module name="RequireFailForTryCatchInJunitCheck"/&gt;
+ * </pre>
+ * @author Richard Veach
+ */
+public class RequireFailForTryCatchInJunitCheck extends AbstractCheck {
+    /**
+     * Violation message key.
+     */
+    public static final String MSG_KEY = "require.fail";
+
+    /**
+     * Fully qualified junit test annotation.
+     */
+    private static final String FQ_JUNIT_TEST = "org.junit.Test";
+    /**
+     * JUnit's fail assertion method name.
+     */
+    private static final String FAIL = "fail";
+
+    /**
+     * {@code true} if the junit test is imported.
+     */
+    private boolean importTest;
+    /**
+     * {@code true} if the junit assert is imported.
+     */
+    private boolean importAssert;
+    /**
+     * {@code true} if the junit fail assertion method is statically imported.
+     */
+    private boolean importStaticFail;
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.IMPORT,
+            TokenTypes.STATIC_IMPORT,
+            TokenTypes.LITERAL_TRY,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void beginTree(DetailAST rootAST) {
+        importTest = false;
+        importAssert = false;
+        importStaticFail = false;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.IMPORT:
+                final String imprt = getImportText(ast);
+
+                if (FQ_JUNIT_TEST.equals(imprt)) {
+                    importTest = true;
+                }
+                if ("org.junit.Assert".equals(imprt)) {
+                    importAssert = true;
+                }
+                break;
+            case TokenTypes.STATIC_IMPORT:
+                final String staticImprt = getImportText(ast);
+
+                if ("org.junit.Assert.fail".equals(staticImprt)) {
+                    importStaticFail = true;
+                }
+                break;
+            case TokenTypes.LITERAL_TRY:
+                examineTry(ast);
+                break;
+            default:
+                Utils.reportInvalidToken(ast.getType());
+                break;
+        }
+    }
+
+    /**
+     * Examines the try block for violations.
+     * @param ast The try block to examine.
+     */
+    private void examineTry(DetailAST ast) {
+        final DetailAST method = getStopParent(ast);
+
+        if (method != null
+                && method.getType() == TokenTypes.METHOD_DEF
+                && ast.findFirstToken(TokenTypes.LITERAL_CATCH) != null
```

Comment:
move "ast.findFirstToken(TokenTypes.LITERAL_CATCH) != null" to inner IF.

and you you see that all expessions that are left could be named as `isTestMethod`, logic will become to simple to read.

Reply:
Done.

Reply:
`method.getType() == TokenTypes.METHOD_DEF` was removed because of the change to `getMethod` which returns null as long as non-methods aren't returned.